tests/unit/plugins/module_utils/test_attachment.py

```diff
@@ -0,0 +1,872 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import os
+import sys
+
+import pytest
+
+from tempfile import mkstemp
+
+from ansible_collections.servicenow.itsm.plugins.module_utils import errors, attachment
+from ansible_collections.servicenow.itsm.plugins.module_utils.client import Response
+
+pytestmark = pytest.mark.skipif(
+    sys.version_info < (2, 7), reason="requires python2.7 or higher"
+)
+
+
+class TestAttachmentGetFileName:
+    def test_name_specified(self):
+        assert (
+            attachment.get_file_name(
+                {
+                    "path": "some/path/file_name.txt",
+                    "name": "attachment_name.txt",
+                    "type": "text/markdown",
+                    "hash": "hash",
+                }
+            )
+            == "attachment_name.txt"
+        )
+
+    def test_name_omitted(self):
+        assert (
+            attachment.get_file_name(
+                {
+                    "path": "some/path/file_name.txt",
+                    "name": None,
+                    "type": "text/markdown",
+                    "hash": "hash",
+                }
+            )
+            == "file_name.txt"
+        )
+
+
+class TestAttachmentGetFileType:
+    def test_type_specified(self):
+        assert (
+            attachment.get_file_type(
+                {
+                    "path": "some/path/file_name.txt",
+                    "name": "attachment_name.txt",
+                    "type": "text/markdown",
+                    "hash": "hash",
```

Comment:
Name and hash fields are not needed here, so we can safely remove them from the test:

```suggestion
                    "path": "some/path/file_name.txt",
                    "type": "text/markdown",
```